syntax = "proto3";

package space_service;

message Vector3f {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Movement {
  Vector3f position = 1;
  Vector3f rotation = 2;
  optional Vector3f velocity = 3;
  optional Vector3f acceleration = 4;
  optional Vector3f angular_velocity = 5;
}

message LoginRequest {
  string username = 1;
}

message LoginReply {
  int32 result = 1;
}

message JoinReply {
  int32 result = 1;
  optional Vector3f position = 2;
}

message AoiPlayer {
  string name = 1;
  Vector3f position = 2;  
  Movement transform = 3;
}

message PlayersEnterSight {
  repeated AoiPlayer players = 1;
}

message PlayersLeaveSight {
  repeated string players = 1;
}

message PlayerMovement {
  string name = 1;
  Movement data = 2;
}

message PlayerMovements {
  repeated PlayerMovement datas = 1;
}

message Ping {
  float t = 1;
}

message Pong {
  float t = 1;
}